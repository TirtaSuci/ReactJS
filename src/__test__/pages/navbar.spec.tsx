import { SessionProvider } from "next-auth/react";
import Navbar from "@/pages/componens/layouts/navbar";
import { render } from "@testing-library/react";

jest.mock("next/router", () => ({
  useRouter: () => ({
    pathname: "/src/pages/componens/layouts/navbar",
    push: jest.fn(),
  }),
}));

describe("Navbar Component", () => {
  it("should render correctly", () => {
    const title = "Mock Title";
    const link = "/mock";

    const { asFragment } = render(
      <SessionProvider session={null}>
        <Navbar title={title} link={link} />
      </SessionProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
