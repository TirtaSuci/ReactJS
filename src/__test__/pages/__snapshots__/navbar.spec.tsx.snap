// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="navbar"
  >
    <a
      href="/mock"
    >
      Mock Title
    </a>
    <div
      class="navbar__profile"
    >
      <button
        class="navbar__pointer"
      >
        Login
      </button>
    </div>
  </div>
</DocumentFragment>
`;
